### PR TITLE
Update 

### PR DESCRIPTION
**Pull request by ashish-narwal**

### CODE REVIEW

File: example_annotate.py [hunk 1]
```diff
@@ -3,13 +3,15 @@
...
 import pkg_resources
 
-from nalaf.utils.readers import TextFilesReader, PMIDReader
+from nalaf.utils.readers import TextFilesReader, PMIDReader, HTMLReader
 from nalaf.utils.readers import StringReader
 from nalaf.utils.writers import ConsoleWriter, TagTogFormat, PubTatorFormat
 from nalaf.structures.dataset_pipelines import PrepareDatasetPipeline
 from nalaf.learning.crfsuite import PyCRFSuite
-from nalaf.domain.bio.gnormplus import GNormPlusGeneTagger
+#from nalaf.domain.bio.gnormplus import GNormPlusGeneTagger
```
Comment (juanmirocks) [inline example_annotate.py]: Just remove it.

File: example_annotate.py [hunk 2]
```diff
@@ -59,7 +66,7 @@
     crf = PyCRFSuite(model_file=pkg_resources.resource_filename('nalaf.data', 'example_entity_model'))
     crf.annotate(dataset, class_id=ENT2_CLASS_ID)
 
-    GNormPlusGeneTagger(ENT1_CLASS_ID, ENTREZ_GENE_ID, UNIPROT_ID).tag(dataset, uniprot=True)
+    #GNormPlusGeneTagger(ENT1_CLASS_ID, ENTREZ_GENE_ID, UNIPROT_ID).tag(dataset, uniprot=True)
```
Comment (juanmirocks) [inline example_annotate.py]: Just remove it.

File: nalaf/download_data.py [hunk 3]
```diff
@@ -13,6 +14,14 @@
 if __name__ == '__main__':
 
     # Download NLTK models
+    try:
+        _create_unverified_https_context = ssl._create_unverified_context
+    except AttributeError:
+        pass
+    else:
+        ssl._create_default_https_context = _create_unverified_https_context
```
Comment (juanmirocks) [inline nalaf/download_data.py]: What is this? Do we really need it?

Do we need it for the downloading of the NLTK models?

File: example_annotate.py [hunk 4]
```diff
@@ -49,7 +51,12 @@
         dataset = PMIDReader(args.pmids).read()
     elif os.path.exists(args.dir_or_file):
         print(warning)
-        dataset = TextFilesReader(args.dir_or_file).read()
+        #dataset = TextFilesReader(args.dir_or_file).read()
```
Comment (juanmirocks) [inline example_annotate.py]: Not good.

We need need ANOTHER subcase (another if else) to differentiate between text file vs. 2 files with .plain.html + .ann.json

File: example_annotate.py [hunk 5]
```diff
@@ -49,7 +51,12 @@
...
+        dataset = HTMLReader(args.dir_or_file, whole_basename_as_docid=True, hdfs_url=None, hdfs_user=None).read()
+        file_html=args.dir_or_file
+        file_json=file_html.replace('.html','.json')
+        AnnJsonAnnotationReader(file_json, read_only_class_id=None, whole_basename_as_docid=True, hdfs_url=None, hdfs_user=None).annotate(dataset)
+
```
Comment (juanmirocks) [inline example_annotate.py]: Don't we need to call the function `validate_entity_offsets` somewhere?

File: nalaf/download_data.py (hunk 3 above)
Comment (juanmirocks) [inline nalaf/download_data.py]: This line is confusing. Any reference or comment that might help the reader?

**Review by juanmirocks — CHANGES_REQUESTED** [review]

File: nalaf/download_data.py (hunk 3 above)
Comment (ashish-narwal) [reply nalaf/download_data.py]: yes, it is required. without this, we have to manually download it and put it to the corresponding directory.

File: example_annotate.py (hunk 2 above)
Comment (ashish-narwal) [reply example_annotate.py]: ok

File: example_annotate.py (hunk 1 above)
Comment (ashish-narwal) [reply example_annotate.py]: ok

File: example_annotate.py (hunk 5 above)
Comment (ashish-narwal) [reply example_annotate.py]: the AnnJsonAnnotationReader calls the validate_entity_offsets itself.

**Review by juanmirocks — APPROVED** [review]
nice

**Comment by juanmirocks** [comment]
Already merged into develop branch.

**Pull request closed without merging**